In [213]:
import os
from os import listdir
from os.path import isdir, join
import numpy as np
mypath = '/nfs/masi/lyui/VBI/CASCIO_CM/'
h = []
ct = []
lgi = []
sci =[]
s = []
sd = []
for root, dirs, files in os.walk(mypath):
    dirs.sort()
    for file in files:
        if file.endswith('lh.mid.reg.H.txt'):
            h.append(join(root, file))
        elif file.endswith('lh.mid.reg.CT.txt'):
            ct.append(join(root, file))
        elif file.endswith('lh.mid.reg.LGI.txt'):
            lgi.append(join(root, file))
        elif file.endswith('lh.mid.reg.SCI.txt'):
            sci.append(join(root, file))
        elif file.endswith('lh.mid.reg.S.txt'):
            s.append(join(root, file))
        elif file.endswith('lh.mid.reg.SD.txt'):
            sd.append(join(root, file))

['/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-127462-x-127462-x-surf_quant_stat_v1-x-c93e71d6-bd77-40f8-8ec3-ff360ce300c9-x-Reg/lh.mid.reg.H.txt',
 '/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-127642-x-127642-x-surf_quant_stat_v1-x-bc38a014-6517-4446-99b7-b297ec2c2ca7-x-Reg/lh.mid.reg.H.txt',
 '/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-127740-x-127740-x-surf_quant_stat_v1-x-58b65f76-c450-4445-ae83-5011f3d5f605-x-Reg/lh.mid.reg.H.txt',
 '/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-208390-x-208390-x-surf_quant_stat_v1-x-54f46b13-f5f5-46ed-b48e-d370e7d7764f-x-Reg/lh.mid.reg.H.txt',
 '/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-208474-x-208474-x-surf_quant_stat_v1-x-746ffd4d-22c9-4e18-a4e4-5cd0949b03b5-x-Reg/lh.mid.reg.H.txt',
 '/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-208697-x-208697-x-surf_quant_stat_v1-x-2512ec54-e600-451d-9886-5bd34a4057ed-x-Reg/lh.mid.reg.H.txt',
 '/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM-x-208832-x-208832-x-surf_quant_stat_v1-x-dce7a77e-b30f-4c60-b26a-d0dc574d5fa9-x-Reg/lh.mid.reg.

In [283]:
print(len(h))

336


In [219]:
list = [h, ct, lgi, sci, s, sd]
for i in range(len(list)):
    if len(list[i]) != 336:
        print("oops")
list[1][1].split("-x-")

['/nfs/masi/lyui/VBI/CASCIO_CM/CASCIO_CM',
 '127642',
 '127642',
 'surf_quant_stat_v1',
 'bc38a014-6517-4446-99b7-b297ec2c2ca7',
 'Reg/lh.mid.reg.CT.txt']

# Print file names in matlab
string = ""
cnt = 0
f = open('/home/haol/Python Script/commands.txt', "w+")
for i in range(len(list)):
    for j in range(len(h)):
        cnt+=1
        string = "/fs4/masi/haol/SurfRemesh -t /fs4/masi/haol/input_mesh_files/sphere_327680.vtk -r /fs4/masi/haol/input_mesh_files/icosphere_7.vtk --outputProperty {} --noheader -p {} &"
        fileNumber = list[i][j].split("-x-")
        f.write(string.format("/fs4/masi/haol/data/"+fileNumber[2],list[i][j])+"\n")
        if cnt%18 == 0:
            f.write("wait" + "\n")
f.close()

In [388]:
import pandas as pd
label = pd.read_csv('/home/haol/Python Script/table.csv')['dx_group'].values
len(label)

221

In [390]:
table = pd.read_csv('/home/haol/Python Script/table.csv')
tab = table['xnat_session_number_Subject'].values
table.tail(4)
table['xnat_session_number_Subject'][217] = '229074'
table['xnat_session_number_Subject'][218] = '229106'
table['xnat_session_number_Subject'][220] = '233853'
table.tail(4)

/home/haol/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.
/home/haol/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """
/home/haol/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


xnat_session_number_Subject  dx_group  gender  age_at_scan  age_at_dx  \
217                     229074.0         0       1           20        NaN   
218                     229106.0         0       0           37        NaN   
219                     235473.0         0       0           32        NaN   
220                     233853.0         1       1           43        NaN   

     Var1                                           Assessor Procstatus    QA  \
217   306  CASCIO_CM-x-235320-x-235320-x-surf_quant_stat_...   COMPLETE  Good   
218   307  CASCIO_CM-x-235430-x-235430-x-surf_quant_stat_...   COMPLETE  Good   
219   309  CASCIO_CM-x-235473-x-235473-x-surf_quant_stat_...   COMPLETE  Good   
220   405  CASCIO_CM-x-Cascio_232708-x-232708b-x-surf_qua...   COMPLETE  Good   

      Session  
217  235320.0  
218  235430.0  
219  235473.0  
220       NaN

In [368]:
data_dir = "/fs4/masi/haol/data/"
file = []
f= open("/fs4/masi/haol/data/H.txt","w+")
good = ['1065', '1078','1065','1384','1491','1506','1511','1652','1691','782','863','894', '232708b']
for filename in sorted(os.listdir(data_dir)):
    if filename.endswith('.H.txt') and not filename.endswith('lh.mid.reg.H.txt'):
        a = filename.replace('.H.txt', '')
        if a in tab or a in good:
            file.append(join(data_dir, filename))
            f.write(join(data_dir, filename)+'\n')
        else:
            print(filename)
f.close()

127642.H.txt
208390.H.txt
208468.H.txt
208474.H.txt
208577.H.txt
208606.H.txt
208689.H.txt
208697.H.txt
208832.H.txt
208963.H.txt
208993.H.txt
20900.H.txt
209097.H.txt
209451.H.txt
209468.H.txt
209560.H.txt
209599.H.txt
209651.H.txt
210643.H.txt
212658.H.txt
215841.H.txt
216549.H.txt
216580.H.txt
227052.H.txt
227219.H.txt
228908.H.txt
232271.H.txt
232555.H.txt
232591.H.txt
233006.H.txt
234047.H.txt
234079.H.txt
234080.H.txt
234094.H.txt
234399.H.txt
234462.H.txt
234596.H.txt
234989.H.txt
235433.H.txt
235529.H.txt


['/fs4/masi/haol/data/1065.H.txt',
 '/fs4/masi/haol/data/1078.H.txt',
 '/fs4/masi/haol/data/127462.H.txt',
 '/fs4/masi/haol/data/127740.H.txt',
 '/fs4/masi/haol/data/130268.H.txt',
 '/fs4/masi/haol/data/1384.H.txt',
 '/fs4/masi/haol/data/1491.H.txt',
 '/fs4/masi/haol/data/1506.H.txt',
 '/fs4/masi/haol/data/1511.H.txt',
 '/fs4/masi/haol/data/1652.H.txt',
 '/fs4/masi/haol/data/1691.H.txt',
 '/fs4/masi/haol/data/208498.H.txt',
 '/fs4/masi/haol/data/209087.H.txt',
 '/fs4/masi/haol/data/209186.H.txt',
 '/fs4/masi/haol/data/209278.H.txt',
 '/fs4/masi/haol/data/209301.H.txt',
 '/fs4/masi/haol/data/209453.H.txt',
 '/fs4/masi/haol/data/209455.H.txt',
 '/fs4/masi/haol/data/209584.H.txt',
 '/fs4/masi/haol/data/209658.H.txt',
 '/fs4/masi/haol/data/209782.H.txt',
 '/fs4/masi/haol/data/209803.H.txt',
 '/fs4/masi/haol/data/209948.H.txt',
 '/fs4/masi/haol/data/209965.H.txt',
 '/fs4/masi/haol/data/210124.H.txt',
 '/fs4/masi/haol/data/210380.H.txt',
 '/fs4/masi/haol/data/210523.H.txt',
 '/fs4/masi/haol/

In [351]:
a = table['Session'].values
len(set(a))

221

In [379]:
with open('/fs4/masi/haol/data/H.txt') as f:  
    dir = f.readlines()
data_dir = [w.replace('\n', '') for w in dir]
data_dir

['/fs4/masi/haol/data/782.H.txt',
 '/fs4/masi/haol/data/863.H.txt',
 '/fs4/masi/haol/data/894.H.txt',
 '/fs4/masi/haol/data/1065.H.txt',
 '/fs4/masi/haol/data/1078.H.txt',
 '/fs4/masi/haol/data/1384.H.txt',
 '/fs4/masi/haol/data/1491.H.txt',
 '/fs4/masi/haol/data/1506.H.txt',
 '/fs4/masi/haol/data/1511.H.txt',
 '/fs4/masi/haol/data/1652.H.txt',
 '/fs4/masi/haol/data/1691.H.txt',
 '/fs4/masi/haol/data/7004.H.txt',
 '/fs4/masi/haol/data/7007.H.txt',
 '/fs4/masi/haol/data/7017.H.txt',
 '/fs4/masi/haol/data/7018.H.txt',
 '/fs4/masi/haol/data/7019.H.txt',
 '/fs4/masi/haol/data/7023.H.txt',
 '/fs4/masi/haol/data/7026.H.txt',
 '/fs4/masi/haol/data/7031.H.txt',
 '/fs4/masi/haol/data/7038.H.txt',
 '/fs4/masi/haol/data/7039.H.txt',
 '/fs4/masi/haol/data/7042.H.txt',
 '/fs4/masi/haol/data/7055.H.txt',
 '/fs4/masi/haol/data/7064.H.txt',
 '/fs4/masi/haol/data/7069.H.txt',
 '/fs4/masi/haol/data/7082.H.txt',
 '/fs4/masi/haol/data/7084.H.txt',
 '/fs4/masi/haol/data/7101.H.txt',
 '/fs4/masi/haol/data/7

In [268]:
H_ = np.array([])
for i in range(len(data_dir)):
    H_ = np.concatenate([H_,np.loadtxt(fname = data_dir[i])]) 
np.shape(H_)

(42762762,)

In [355]:
#a = np.loadtxt(fname = data_dir[1])
H_[1:5, 2:4]
H_.dtype

dtype('float64')

# Make matrices
CT = np.array([])
for i in range(0, len(ct)):
    CT = np.concatenate([CT,np.loadtxt(fname = ct[i])])  

LGI = np.array([])
for i in range(0, len(lgi)):
    LGI = np.concatenate([LGI,np.loadtxt(fname = lgi[i])])  

SCI = np.array([])
for i in range(0, len(sci)):
    SCI = np.concatenate([SCI,np.loadtxt(fname = sci[i])]) 

S = np.array([])
for i in range(0, len(s)):
    S = np.concatenate([S,np.loadtxt(fname = s[i])])  

SD = np.array([])
for i in range(0, len(sd)):
    SD = np.concatenate([SD,np.loadtxt(fname = sd[i])])

In [244]:
for i in range(len(h)):
    if np.shape(np.loadtxt(fname = h[i])) != (163842,):
        print(i)
a=np.loadtxt(fname = h[2])
np.shape(a)
#np.shape(np.reshape(H,163824,-1))
#np.shape(CT)

(163842,)

In [281]:
H_ = np.reshape(H_, (-1,163842))
np.shape(H_)
H[:, :]

(261, 163842)

#hm= open("/fs4/masi/haol/data/Hmat.txt","w+")
np.savetxt("/fs4/masi/haol/data/Hmat.txt", H_)

In [288]:
label

array([0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0,
       1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0,
       1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0,
       0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1,
       0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1,
       1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1,
       1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1,
       1])

In [378]:
'233853' in b

True

In [331]:
filename = '233853.H.txt'
a = filename.replace('.H.txt', '')
if a in b:
    print('Yay')

Yay


In [214]:
a = [[2,3],[2,4]]
a
np.shape(a)
#np.shape(np.expand_dims(a,1))
a[1][1]

4

In [367]:
H_= H_.astype(np.float32)
H_.dtype

dtype('float32')

In [377]:
import re

def atoi(text):
    return int(text) if text.isdigit() else text

def natural_keys(text):
    return [ atoi(c) for c in re.split(r'(\d+)', text) ]

file.sort(key=natural_keys)

In [669]:
file

['/fs4/masi/haol/data/782.H.txt',
 '/fs4/masi/haol/data/863.H.txt',
 '/fs4/masi/haol/data/894.H.txt',
 '/fs4/masi/haol/data/1065.H.txt',
 '/fs4/masi/haol/data/1078.H.txt',
 '/fs4/masi/haol/data/1384.H.txt',
 '/fs4/masi/haol/data/1491.H.txt',
 '/fs4/masi/haol/data/1506.H.txt',
 '/fs4/masi/haol/data/1511.H.txt',
 '/fs4/masi/haol/data/1652.H.txt',
 '/fs4/masi/haol/data/1691.H.txt',
 '/fs4/masi/haol/data/7004.H.txt',
 '/fs4/masi/haol/data/7007.H.txt',
 '/fs4/masi/haol/data/7017.H.txt',
 '/fs4/masi/haol/data/7018.H.txt',
 '/fs4/masi/haol/data/7019.H.txt',
 '/fs4/masi/haol/data/7023.H.txt',
 '/fs4/masi/haol/data/7026.H.txt',
 '/fs4/masi/haol/data/7031.H.txt',
 '/fs4/masi/haol/data/7038.H.txt',
 '/fs4/masi/haol/data/7039.H.txt',
 '/fs4/masi/haol/data/7042.H.txt',
 '/fs4/masi/haol/data/7055.H.txt',
 '/fs4/masi/haol/data/7064.H.txt',
 '/fs4/masi/haol/data/7069.H.txt',
 '/fs4/masi/haol/data/7082.H.txt',
 '/fs4/masi/haol/data/7084.H.txt',
 '/fs4/masi/haol/data/7101.H.txt',
 '/fs4/masi/haol/data/7

In [374]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/H.txt","w+")
for filename in file:
            f.write(join(data_dir, filename)+'\n')
f.close()

# Training, testing, and validation data selection (**)
import random
CT_train = CT["Session"].values 
random.shuffle(CT_train)
a = random.choices(CT_train, k=45)
ASD_train = ASD["Session"].values 
random.shuffle(ASD_train)
b = random.choices(ASD_train, k=42)

In [852]:
tb = pd.read_csv('/home/haol/Python Script/updated_table.csv')
CT = tb[tb['dx_group']==0]
ASD = tb[tb['dx_group']==1]
tb['Session']=tb['Session'].values.astype(np.int64)
print("CT:"+str(len(CT)) + " " + "ASD:"+str(len(ASD)))

CT:145 ASD:111


In [701]:
CT_train = CT["Session"].values
CT_train.sort()
CT_train = CT_train[:119]
ASD_train = ASD["Session"].values 
ASD_train.sort()
ASD_train = ASD_train[:90]
len(CT_train)

119

In [702]:
## Shuffled lists
newT = pd.DataFrame(data=CT_train,columns=['Session'])  
newT['dx_group'] = CT["dx_group"].values[:119]
newT2 = pd.DataFrame(data=ASD_train, columns=['Session'])  
newT2['dx_group'] = ASD["dx_group"].values[:90]
frames = [newT, newT2]
T = pd.concat(frames, ignore_index=True)
T = T.sample(frac=1).reset_index(drop=True) # output T to csv file and print out the session number as path into a txt file named as 'Htrain.txt'
trainfile = T['Session'].values.astype(str)
T

Session  dx_group
0     227330         1
1       7026         1
2     215700         0
3     215614         1
4       7068         0
5     227546         0
6     227744         0
7     227232         1
8     209948         0
9     227602         1
10    216724         1
11      7112         0
12    213923         1
13    210888         0
14      7071         0
15    127462         0
16      7106         0
17    209560         0
18      1043         0
19      7085         0
20    226538         0
21    208606         1
22    216598         1
23    209782         1
24      1037         0
25    227063         1
26    214077         0
27    227537         0
28    210889         0
29    214255         0
..       ...       ...
179     7048         1
180   227112         0
181   228482         0
182     7055         1
183     7082         1
184   208498         1
185     1065         1
186   227441         1
187   216721         1
188   209451         0
189   226958         0
190   210762         0
191   226509         0
192   209803         0
193   211584         1
194   226654         0
195   209301         1
196   227442         0
197   212481         0
198   212383         1
199     7104         1
200   209658         0
201   211136         0
202   209965         0
203     7074         0
204   228623         0
205   212988         0
206   216233         1
207     1532         0
208   214032         0

[209 rows x 2 columns]

In [709]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/LGItrain.txt","w+")
for filename in trainfile:
            f.write(join(data_dir, filename) + '.LGI.txt'+'\n')
f.close()
T.to_csv ('/home/haol/Python Script/LGItraintable.csv', index = None, header=True)

In [718]:
CT_test = CT["Session"].values
CT_test.sort()
CT_test = CT_test[119:134]
ASD_test = ASD["Session"].values 
ASD_test.sort()
ASD_test = ASD_test[90:102]
newTest = pd.DataFrame(data=CT_test,columns=['Session'])  
newTest['dx_group'] = CT["dx_group"].values[:15]
newTest2 = pd.DataFrame(data=ASD_test, columns=['Session'])  
newTest2['dx_group'] = ASD["dx_group"].values[:12]
frames = [newTest, newTest2]
Test = pd.concat(frames, ignore_index=True)
Test = Test.sample(frac=1).reset_index(drop=True) # output T to csv file and print out the session number as path into a txt file named as 'Htrain.txt'
testfile = Test['Session'].values.astype(str)
Test

Session  dx_group
0    227851         1
1    235057         0
2    229006         0
3    228612         1
4    227853         1
5    229106         0
6    230643         0
7    229235         0
8    229074         0
9    228996         0
10   228458         1
11   229177         0
12   229024         0
13   235022         0
14   229076         0
15   227644         1
16   227698         1
17   228161         1
18   232573         0
19   235104         0
20   227790         1
21   234953         0
22   230380         1
23   227950         1
24   232343         0
25   230910         1
26   228027         1

In [724]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/SCItest.txt","w+")
for filename in testfile:
            f.write(join(data_dir, filename) + '.SCI.txt'+'\n')
f.close()
Test.to_csv ('/home/haol/Python Script/SCItesttable.csv', index = None, header=True)

In [725]:
CT_vad = CT["Session"].values
CT_vad.sort()
CT_vad = CT_vad[134:]
ASD_vad = ASD["Session"].values 
ASD_vad.sort()
ASD_vad = ASD_vad[102:]
newvad = pd.DataFrame(data=CT_vad,columns=['Session'])  
newvad['dx_group'] = CT["dx_group"].values[:15]
newvad2 = pd.DataFrame(data=ASD_vad, columns=['Session'])  
newvad2['dx_group'] = ASD["dx_group"].values[:11]
frames = [newvad, newvad2]
Vad = pd.concat(frames, ignore_index=True)
Vad = Vad.sample(frac=1).reset_index(drop=True) # output T to csv file and print out the session number as path into a txt file named as 'Htrain.txt'
vadfile = Vad['Session'].values.astype(str)
Vad

Session  dx_group
0    234404         1
1    235186         0
2    234613         1
3    230975         1
4    233682         1
5    235301         0
6    235166         0
7    233302         1
8    235155         0
9    235181         0
10   235231         0
11   235320         0
12   235473         0
13   235279         0
14   233618         1
15   234390         1
16   235222         0
17   235249         0
18   232768         1
19   232224         1
20   235198         0
21   232708         1
22   233853         1
23   235430         0
24   235220         0
25   235212         0

In [731]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/CTvalidation.txt","w+")
for filename in vadfile:
            f.write(join(data_dir, filename) + '.CT.txt'+'\n')
f.close()
Vad.to_csv ('/home/haol/Python Script/CTvalidationtable.csv', index = None, header=True)

## Matrix Dimension Testing

In [1011]:
DDD=[[[9]*5]*2]*3

In [1010]:
a = [1, 2, 3, 4, 5]
b = [6, 7, 8, 9, 10]
c = [11, 12, 13, 14, 15]
DDD = np.concatenate((a,b,c))
DDD = np.reshape(DDD,(3, -1))
#print(DDD)
#print(DDD[:3, :2])
#DDD.T[:3, :2]

In [ ]:
import torch 
output = torch.tensor(
  [
      [3, 1, 0],
      [3, 7, 2],
      [3, 8, 9],
      [7, 6, 4],
      [2, 7, 5],
      [6, 6, 8]
  ]
)
output.max(1, keepdim=True)[0]

## Filtering files for remashing

In [805]:
with open('//home/haol/Python Script/commands.txt') as f:  
    com = f.readlines()
uniNum = []
k = []
cnt = 0
for i in range(len(com)):
    if com[i] != 'wait\n':
        cnt += 1
        k = com[i].split("-x-")
        if cnt % 6 == 0:
            uniNum.append(k[2])

str

uniNum[219] = ''
uniNum.pop(219)

if tb['Session'].values.astype(int)[i] in uni:
        print(tb['Session'].values.astype(int)[i])
        continue 
    else:
        missing.append(str(tb['Session'].values.astype(int)[i]))

In [838]:
uni = [int(i) for i in uniNum]
missing = []
for i in range(len(uni)):
    if tb['Session'].values.astype(int)[i] not in uni:
        missing.append(tb['Session'].values.astype(int)[i])
len(missing)
#missing 

42

In [846]:
name = []

for i in range(len(missing)):
    name.extend(tb[tb['Session'] ==  missing[i]]['Assessor'].values)

In [850]:
mi = pd.DataFrame(data=missing,columns=['Session'])  
mi['Assessor'] = name
mi.to_csv ('/home/haol/Python Script/missing_session.csv', index = None, header=True)

# Make ROC

In [18]:
import pandas as pd
tb = pd.read_csv('/home/haol/Python Script/updated_table.csv')
frames = []
#CT = CT.sample(frac=1).reset_index(drop=True)
#ASD = ASD.sample(frac=1).reset_index(drop=True)
for i in range(10):
    newTest = pd.DataFrame(data=CT[14*i:14*(i+1)],columns=['Session'])  
    newTest['dx_group'] = CT["dx_group"].values[:14]
    newTest2 = pd.DataFrame(data=ASD[11*i:11*(i+1)], columns=['Session'])  
    newTest2['dx_group'] = ASD["dx_group"].values[:11]
    frames.append([newTest, newTest2])

NameError: name 'CT' is not defined

In [954]:
newTest = pd.DataFrame(data=CT[140:],columns=['Session'])  
newTest['dx_group'] = CT["dx_group"].values[:5]
newTest2 = pd.DataFrame(data=ASD[110:], columns=['Session'])  
newTest2['dx_group'] = ASD["dx_group"].values[:1]
frames.append([newTest, newTest2])

In [955]:
Table = []
for i in range(11):
    T2 = pd.concat(frames[i], ignore_index=True)
    #T2 = T2.sample(frac=1).reset_index(drop=True)
    Table.append(T2)

In [956]:
ShuffledT = pd.concat(Table, ignore_index=True)
Train = ShuffledT[:205]
Test = ShuffledT[205:230]
Valid = ShuffledT[230:]
trainfile = Train['Session'].values.astype(str)
testfile = Test['Session'].values.astype(str)
validfile = Valid['Session'].values.astype(str)

In [973]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/CV3/CTtrain.txt","w+")
for filename in trainfile:
            f.write(join(data_dir, filename) + '.CT.txt'+'\n')
f.close()
Train.to_csv ('/home/haol/Python Script/CV3/CTtraintable.csv', index = None, header=True)

In [974]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/CV3/CTtest.txt","w+")
for filename in testfile:
            f.write(join(data_dir, filename) + '.CT.txt'+'\n')
f.close()
Test.to_csv ('/home/haol/Python Script/CV3/CTtesttable.csv', index = None, header=True)

In [975]:
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/CV3/CTvalidation.txt","w+")
for filename in validfile:
            f.write(join(data_dir, filename) + '.CT.txt'+'\n')
f.close()
Valid.to_csv ('/home/haol/Python Script/CV3/CTvalidationtable.csv', index = None, header=True)

## Better_Loader

In [8]:
import pandas as pd
tb = pd.read_csv('/home/haol/Python_Script/updated_table.csv')
TD = tb[tb['dx_group']==0][:111]
ASD = tb[tb['dx_group']==1]
TD = TD.sample(frac=1).reset_index(drop=True)
ASD = ASD.sample(frac=1).reset_index(drop=True)

In [9]:
print(str(len(ASD)) +" "+ str(len(TD)))

111 111


In [10]:
k = 6
S_ASD = int(len(ASD)/k)
S_TD = int(len(TD)/k)

In [11]:
import numpy as np
P_ASD = []
P_TD = []
for i in range(k+1):
    P_ASD.append(i*S_ASD)
    P_TD.append(i*S_TD)
#P_TD
P_ASD
#P_TD

[0, 18, 36, 54, 72, 90, 108]

In [12]:
#P_ASD[-1] = 111
P_ASD
CV = 0
P_TD

[0, 18, 36, 54, 72, 90, 108]

In [13]:
def k_folds(CV,TD,ASD,size_v=1,size_t=1):
    frames = []
    frames2 = []
    frames3 = []
    for i in range(k-size_v-size_t):
        idx1 = (CV+i)%k
        idx2 = idx1+1
        a = ASD.loc[range(P_ASD[idx1],P_ASD[idx2])]
        b = TD.loc[range(P_TD[idx1],P_TD[idx2])]
        frames.append(a)
        frames.append(b)
    train = pd.concat(frames, ignore_index=False)
    for i in range(size_t):
        idx3 = (idx2+i)%k
        idx4 = idx3+1
        c = ASD.loc[range(P_ASD[idx3],P_ASD[idx4])]
        d = TD.loc[range(P_TD[idx3],P_TD[idx4])]
        frames2.append(c)
        frames2.append(d)
    test = pd.concat(frames2, ignore_index=False)
    for i in range(size_v):
        idx5 = (idx4+i)%k
        idx6 = idx5+1
        e = ASD.loc[range(P_ASD[idx5],P_ASD[idx6])]
        f = TD.loc[range(P_TD[idx5],P_TD[idx6])]
        frames3.append(e)
        frames3.append(f)
    valid = pd.concat(frames3, ignore_index=False)
    return train, test, valid

In [14]:
train, test, valid = k_folds(0, TD, ASD)
# trainfile = train['Session'].values.astype(str)
# testfile = test['Session'].values.astype(str)
# validfile = valid['Session'].values.astype(str)

In [15]:
train

scan_number_Subject  dx_group  phi_gender  age_at_scan  age_at_dx  \
0              210380.0         1           1           27       23.0   
1              215605.0         1           1           35        2.0   
2                7018.0         1           1            7        NaN   
3              215271.0         1           1           10       10.0   
4                4034.0         1           1           12       10.0   
5              227359.0         1           1            7        3.0   
6              227582.0         1           1           10        9.0   
7                7042.0         1           1            6        NaN   
8              227790.0         1           1           23        NaN   
9              227063.0         1           1            7        5.0   
10             226999.0         1           1            7        4.0   
11                  NaN         1           0           17        6.0   
12             213007.0         1           0            8        4.0   
13             234390.0         1           1           22        NaN   
14             213923.0         1           1           20       20.0   
15             216724.0         1           0            8        3.0   
16             212519.0         1           0            8        4.0   
17             209186.0         1           0           14       10.0   
0                7090.0         0           1            7        NaN   
1              227371.0         0           1           22        NaN   
2              209584.0         0           1           15        NaN   
3              228519.0         0           1            9        NaN   
4              215700.0         0           0           23        NaN   
5              228429.0         0           1           10        NaN   
6                4033.0         0           1           11        NaN   
7              209453.0         0           0           26        NaN   
8              210888.0         0           0            9        NaN   
9              209948.0         0           1           24        NaN   
10             212929.0         0           1           16        NaN   
11             212533.0         0           1            9        NaN   
..                  ...       ...         ...          ...        ...   
60               4062.0         1           1           19        7.0   
61             214092.0         1           1           20       20.0   
62               7038.0         1           1            5        3.0   
63             233302.0         1           0           24        2.0   
64             226959.0         1           1            8        3.0   
65             232708.0         1           1           43       34.0   
66             228027.0         1           1           26       15.0   
67               7109.0         1           0            7        4.0   
68             215346.0         1           0           10        4.0   
69             226677.0         1           0           30        NaN   
70               7084.0         1           1            5        2.0   
71             212605.0         1           1           23        7.0   
54               4066.0         0           1           10        NaN   
55             214077.0         0           0            8        NaN   
56               7112.0         0           1            5        NaN   
57             209965.0         0           1           34        NaN   
58             211135.0         0           1            8        NaN   
59             214176.0         0           0            8        NaN   
60             227169.0         0           1           26        NaN   
61             214184.0         0           1           10        NaN   
62             227662.0         0           0           24        NaN   
63             227744.0         0           1           32        NaN   
64             211120.0         0           1           13    

from os.path import isdir, join
data_dir = "/fs4/masi/haol/data/"
f= open("/fs4/masi/haol/data/CV6/CTtrain.txt","w+")
for filename in trainfile:
            f.write(join(data_dir, filename) + '.CT.txt'+'\n')
f.close()
train.to_csv ('/home/haol/Python Script/CV6/CTtraintable.csv', index = None, header=True)
g= open("/fs4/masi/haol/data/CV6/CTtest.txt","w+")
for filename in testfile:
            g.write(join(data_dir, filename) + '.CT.txt'+'\n')
g.close()
test.to_csv ('/home/haol/Python Script/CV6/CTtesttable.csv', index = None, header=True)
h= open("/fs4/masi/haol/data/CV6/CTvalidation.txt","w+")
for filename in validfile:
            h.write(join(data_dir, filename) + '.CT.txt'+'\n')
h.close()
valid.to_csv ('/home/haol/Python Script/CV6/CTvalidationtable.csv', index = None, header=True)

lst = []
def r():
    a = 22
    b = 33
    c = 55
    return a,b,c
for i in range(3):
    lst.append(r())
print(lst)
lst[0][1]
lst[0][0]

In [276]:
a=[[[1],[2],[3]]]
a.append([[4],[5],[6]])
a.append([[7],[8],[9]])
np.shape(a) #(2,3,1)
# a[0][0][0] = 3
# a[0][2][0] = 4
# a[0][3][0] = 5
# a[0][1][0] = 6
# #a[0][1][3] = 4
#a[0][1][4] = 5
a[2][:2]
#len(a[0])

[[7], [8]]

In [285]:
k = []
for i in range(len(a[0])):
    k.extend(a[i][:2])
k

[[1], [2], [4], [5], [7], [8]]

In [315]:
data_dir = ['/fs4/masi/haol/data/7038.lh.CT.txt','/fs4/masi/haol/data/7039.lh.CT.txt',]
x = [None]*2
# for i in range(163824):
#     x[i] = i
for i in range(2):
    x[i] = np.loadtxt(fname = data_dir[i])
x = np.reshape(x, (-1, 163842))
np.shape(x)
x

array([[2.78554 , 3.938169, 3.010417, ..., 1.950052, 3.960628, 3.586583],
       [4.561488, 3.683042, 2.009675, ..., 1.88325 , 2.784101, 2.330043]])

In [316]:
data_dir = ['/fs4/masi/haol/data/7038.lh.CT.txt','/fs4/masi/haol/data/7039.lh.CT.txt',]
H_ = np.array([])
for i in range(2):
    H_ = np.concatenate([H_,np.loadtxt(fname = data_dir[i])]) 
np.shape(H_)
H_ = np.reshape(H_, (-1, 163842))
H_
#np.shape(H_)

array([[2.78554 , 3.938169, 3.010417, ..., 1.950052, 3.960628, 3.586583],
       [4.561488, 3.683042, 2.009675, ..., 1.88325 , 2.784101, 2.330043]])

In [350]:
a = valid.replace({'Scan': 'Improved 3D'}, 1)
a = a.replace({'Scan': 'Improved 3D SENSE'}, 2)
a = a.replace({'Scan': 'T1W'}, 3)
a = a.replace({'Scan': 'T1W/3D/TFE'}, 4)
a['Scan'].values.astype(np.int64)

array([1, 4, 1, 3, 4, 1, 2, 1, 1, 1, 2, 2, 3, 2, 3, 2, 1, 3, 3, 4, 2, 1,
       4, 3, 3, 3, 2, 4, 4, 1, 1, 4, 1, 1, 2, 2])

In [366]:
wal = pd.read_csv('/nfs/masi/lyui/WALLACE/demo_wallace_full.csv')
wal.rename(columns={'phi_dxgroup':'dx_group'}, inplace=True)
k = pd.concat([tb, wal], ignore_index=False)
a = wal['Session'].values

In [376]:
cnt = 0
for i in a:
     if i in [222, 236972]:
        cnt +=1
        print(i)
print(cnt)    

236972
1


In [1]:
len(a)

NameError: name 'a' is not defined